mbedtls/src/ssl/context.rs

```diff
@@ -233,7 +233,7 @@ impl<T> Context<T>  {
     // This function ultimately ensure the semantics:
     // Returned value `Ok(n)` always means n bytes of data has been sent into c-mbedtls's buffer (some of them might be sent out through underlying IO)
     pub(super) fn async_write(&mut self, buf: &[u8]) -> Result<usize> {
-        while self.handle().out_left > 0 {
+        while self.handle().private_out_left > 0 {
             self.flush_output()?;
```

Comment:
Since https://github.com/Mbed-TLS/mbedtls/issues/4183 is not been fixed in `3.4.0 mbedtls`.

I just moved the  `mbedtls_ssl_flush_output` from private header into public header.

Reply:
Instead of messing with the headers, let's just write the `extern "C"` block for this one function manually (in mbedtls, not mbedtls-sys)

Reply:
fixed